# Determiant_of_integration
Determiant_of_integration includes R scripts that process data in the same way as in Matrougui et al. (2025).

It is composed of several scripts, located in `src/`, to run in the same order as listed here. The scripts process the number of integrations and sequencing depth specifically to determine the factors influencing genomic polydnavirus integration.

In `examples/`, you will find examples of files needed to run the scripts. Examples are provided for the number of integrations and sequencing depth of host-wasp systems involving four species of Cotesia wasps and five species of lepidopteran hosts. All of this data was produced using Chimera https://github.com/HeloiseMuller/Chimera and WorkflowBowBlast https://github.com/HeloiseMuller/WorkflowBowBlast pipelines.

# 1_merged_chimera_depth.R
Chimera and Depth Data Processing
### Input
This script requires two main input files, each representing merged data from all analyzed samples.

1. File: all_chimera.csv : Chimera Data

This file contains information about chimeric sequences detected along HIM segments.
The original data came from files named `${samples}_all_chimera_alongHIMs_IE.txt`, which were generated through a two-step process:

  Step 1: Execution of the WorkflowBowBlast pipeline https://github.com/HeloiseMuller/WorkflowBowBlast with the blastn=TRUE option enabled. This step produces BLASTn alignments.
  
  Step 2: Use of the script `3-FindChimera_alongSegments.R` part of the Chimera pipeline. This script processes the BLASTn results to identify chimeric reads along HIM segments.
  
The file all_chimera.csv is a merged version of the outputs from all samples.

2. File: all_depth.csv : Sequencing Depth Data
 
This file contains depth of coverage information for all samples, aggregated along specific segments.
The original data were stored in files named `${samples}_coverage_positions`, generated using the CovWindows software https://github.com/HeloiseMuller/CovWindows.
The coverage input files used by CovWindows were previously generated by the WorkflowBowBlast pipeline, using the options bowtie2=TRUE and coverage=TRUE.

### Ouput
The script generates one output files the the merger of chimera and depth Data: `chimera_depth.txt`

# 2_BUSCO_Depth_and_Integration_Analysis.R
BUSCO-Normalized Depth Analysis of Integration Events

This R script performs depth normalization and  analysis of polydnavirus circle integrations in host genomes, using BUSCO gene coverage as a reference for sequencing depth. It is designed to produce normalize and filter the data for visualization and Bayesian analysis.

### Input
1. BUSCO Coverage Data
   
`*_BUSCO_coverage_positions.txt`: Host BUSCO coverage for each sample.
`*_BUSCO_wasp_coverage_positions.txt`: Wasp BUSCO coverage for each sample.

These files are generated using:
BUSCO v5.7.1 (insect_odb10 dataset) and CovWindows https://github.com/HeloiseMuller/CovWindows

2. Depth and Integration Data
   
`chimera_depth.txt`: Integration and depth data per segment and sample, generated by `1_merged_chimera_depth.R`.

### Output

The script generates two main output files:

`chimera_depth_Bayes.txt`	Merged dataset ready for Bayesian statistical analysis
`Chim_Int.txt` normalized data ready for plotting









